### DISCUSSION TITLE
About complete Custom External UI

### DISCUSSION THREAD

**Discussion by bhagatus** [oleg-shilo/wixsharp#1719] [General]
Hello,

This is about complete custom external UI (https://github.com/oleg-shilo/wixsharp/wiki/Deployment-scenarios#complete-custom-external-ui)
As I understand correctly, we need to have MSI for this approach and after that we will wire up events with the external UI.
Please correct me if I am wrong.

**Comment by oleg-shilo** [comment]
Yes, you understood it right. External UI (as well as the stock UI) are there only to collect user input and then use it to execute the MSI.
IE user selects features to be installed, then these feature names are joined with the coma and passed to the MSI being executed as public property `ADDLOCAL`. 

Stock UI does it via MSI Windows service. External UI does the same but in the external UI app itself.

**Reply by bhagatus** [reply]
Thanks for the clarification.
